google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/AddCloudLibrariesDialog.java

```diff
@@ -126,6 +133,29 @@ protected void doOKAction() {
     }
   }
 
+  /**
+   * Fetches all the available roles for the {@link CloudProject} and returns the set of roles
+   * corresponding to the selected client libraries.
+   *
+   * @param apis the set of {@link CloudLibrary apis} selected
+   * @return the set of {@link Role roles} corresponding to the selected apis
+   */
+  private Set<Role> getServiceAccountRolesRoles(Set<CloudLibrary> apis) {
+    List<Role> serviceAccountRoles = CloudApiManager.getServiceAccountRoles(getCloudProject());
```

Comment:
Looks like we have to get a complete set of roles for the user and we only need a smaller subset of those to  get list of roles for selected APIs? I briefly looked myself too and couldn't find anything more specific, so it seems it's currently not possible to obtain list of roles per API/resource?

Reply:
I couldn’t find anything either.

Reply:
Looks like I need to fix the name of that method too :)